downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,460 @@
+
+## Runtime updates
+
+### Improvements
+
+#### Strand dump showing the total strand count
+
+The total number of strand groups and strands created in a Ballerina program is now shown in the strand dump.
+
+#### Allow ambiguous target types in `fromJsonWithType` and `cloneWithType`
+
+The `fromJsonWithType` and `cloneWithType` functions returned an error in the earlier implementation when the user-specified target type was a union type consisting of multiple convertible types for a given source value. Now, for such user-specified target types, the operation succeeds without returning an error by returning a value whose inherent type is as follows. If the given source value belongs to at least one of the member types of the target union type, the inherent type of the returned value is the first found type to which the source value belongs. Otherwise, the inherent type of the returned value is the first convertible type in the union.
+
+For example, the operations below will not return an error.
+
+```ballerina
+type PetByAge record {
+    int age;
+    string nickname?;
+};
+
+type PetByType record {
+    "Cat"|"Dog" pet_type;
+    boolean hunts?;
+};
+
+type Pet PetByAge|PetByType;
+
+public function main() {
+    json j = {
+        "nickname": "Fido",
+        "pet_type": "Dog",
+        "age": 4
+    };
+    Pet|error pet = j.fromJsonWithType(); // inherent type of pet will be PetByAge
+
+    j = {x: 1};
+    record {float x;}|record {int x;}|error v = j.cloneWithType(); // inherent type of v will be record {int x;}
+}
+```
+
+#### Configurable support for the array of tables
+
+The configurable feature is improved to support the array of tables using a two-dimensional TOML array of in-line tables.
+
+For example, if the configurable variables are defined in the following way,
+
+```ballerina
+type Person record {
+    readonly int id;
+    string name;
+};
+
+configurable table<map<int>>[] array1 = ?;
+configurable table<Person> key (id)[] array2 = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+```toml
+array1 = [[{a = 1, b = 2}, {c = 3, d = 4}], [{e = 5, f = 6}]]
+
+array2 = [[{id = 1, name = "Anne"}, {id = 2, name = "Bob"}], [{id = 3, name = "Charles"}]]
+
+```
+
+#### Configurable support for the `nil` type
+
+The configurable feature is now improved to support the following `()` cases.
+
+- configurable variables of `nil` type 
+The configurable variables of the `nil` type are now supported if the configuration is optional.
+
+For example, the following will be initialized with the default values without providing a compilation error.
+
+```ballerina
+configurable () nilValue = ();
+configurable ())[] nilArray = [];
+```
+
+- record fields with `nil` type
+The record fields of `nil` type or a union of `nil` are now supported by the configurable feature only if the field contains a default value.
```

Comment:
The below text adn code should be indented to match with bullet.

Reply:
Done.